python/tests/test_highlevel.py

```diff
@@ -4261,3 +4261,23 @@ def test_macs_error(self):
             ValueError, match="macs output only supports single letter alleles"
         ):
             ts.to_macs()
+
+
+class TestTreeSequenceGetSite:
+    """
+    Tests for getting Site objects from a TreeSequence object
+    by specifying site id or position.
+    """
+
+    def get_example():
+        tables = tskit.TableCollection(sequence_length=10)
+        tables.sites.add_row(position=3, ancestral_state="A")
+        tables.sites.add_row(position=5, ancestral_state="C")
+        tables.sites.add_row(position=7, ancestral_state="G")
+        return tables.tree_sequence()
+
+    @pytest.mark.parametrize("position", [3.0, 5.0, 7.0])
+    def test_site_position(self, position):
+        ts = self.example()
+        site = ts.site(position=position)
+        assert site.position == position
```

Comment:
This is a good start. So we also need:
```
def test_site_id_and_position(self):
    pass

def test_no_args(self):
    pass

@pytest.mark.parametrize(...)
def test_position_bad_type(self, bad_type):
    pass

@pytest.mark.parametrize("position", [-1, 10, 11]):
def test_position_out_of_bounds(self, position):
    pass

@pytest.mark.parametrize("position", [0, 2.999999, 3.000001, 4.9999, 5.00001, 9.99999]):
def test_bad_position(self, position):
    pass

Reply:
It should only work for discrete positions, not continuous?

Reply:
Could I do this to check whether ``position`` holds an integer?

```
a = 9.4
int(a) == a
a = 3.0
int(a) == a
```

Reply:
NVM. I think I misunderstood ``test_bad_position()``.